codex/rest/api.nim

```diff
@@ -157,12 +171,29 @@ proc initDataApi(node: CodexNodeRef, repoStore: RepoStore, router: var RestRoute
       #
       await request.handleExpect()
 
+      let mimetype: string = request.headers.getString(ContentTypeHeader)
+
+      if mimetype != "":
+        var m = newMimetypes()
+        let extension = m.getExt(mimetype, "")
+        if extension == "":
+            return RestApiResponse.error(Http422, "The MIME type is not valid.")
+
+      const ContentDispositionHeader = "Content-Disposition"
+      let contentDisposition = request.headers.getString(ContentDispositionHeader)
+      let filename = getFilenameFromContentDisposition(contentDisposition)
+
+      if filename != "" and not isValidFilename(filename):
```

Comment:
Interesting, where is this coming from? Is it OS-specific? Will it cause issues when trying to download the file to another OS if the file contains invalid characters?

Reply:
It is coming for Nim os package. It is not OS-specific here is the description: 

> Returns true if filename is valid for crossplatform use.
This is useful if you want to copy or save files across Windows, Linux, Mac, etc. It uses invalidFilenameChars, invalidFilenames and maxLen to verify the specified filename.

| Will it cause issues when trying to download the file to another OS if the file contains invalid characters?

The idea is to prevent invalid names from being accepted across platforms to avoid situations like this.

Reply:
Oh OK great stuff.